Sources/NIO/SelectorEpoll.swift

```diff
@@ -0,0 +1,290 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// Represents the `epoll` filters/events we might use:
+///
+///  - `hangup` corresponds to `EPOLLHUP`
+///  - `readHangup` corresponds to `EPOLLRDHUP`
+///  - `input` corresponds to `EPOLLIN`
+///  - `output` corresponds to `EPOLLOUT`
+///  - `error` corresponds to `EPOLLERR`
+private struct EpollFilterSet: OptionSet, Equatable {
+    typealias RawValue = UInt8
+
+    let rawValue: RawValue
+
+    static let _none = EpollFilterSet([])
+    static let hangup = EpollFilterSet(rawValue: 1 << 0)
+    static let readHangup = EpollFilterSet(rawValue: 1 << 1)
+    static let input = EpollFilterSet(rawValue: 1 << 2)
+    static let output = EpollFilterSet(rawValue: 1 << 3)
+    static let error = EpollFilterSet(rawValue: 1 << 4)
+
+    init(rawValue: RawValue) {
+        self.rawValue = rawValue
+    }
+}
+
+extension EpollFilterSet {
+    /// Convert NIO's `SelectorEventSet` set to a `EpollFilterSet`
+    init(selectorEventSet: SelectorEventSet) {
+        var thing: EpollFilterSet = [.error, .hangup]
+        if selectorEventSet.contains(.read) {
+            thing.formUnion(.input)
+        }
+        if selectorEventSet.contains(.write) {
+            thing.formUnion(.output)
+        }
+        if selectorEventSet.contains(.readEOF) {
+            thing.formUnion(.readHangup)
+        }
+        self = thing
+    }
+}
+
+#if os(Linux) || os(Android)
```

Comment:
can't we now just do one huge `#if os(Linux) || os(Android)` around the whole file?

Reply:
Yep, should do it like that.

Reply:
Fixed in e19e685.